provisioning/image_customization.go

```diff
@@ -0,0 +1,233 @@
+			Name: name,
+		}
+	}
+}
+
+func setImagePublishUrl(imageCustomizationService string, imagesBindAddress string, namespace string) corev1.EnvVar {
+	imageCustomizationServiceName := imageCustomizationService + "." + namespace + "." + "svc.cluster.local"
+	return corev1.EnvVar{
+		Name:  imagesPublishAddress,
+		Value: "http://" + imageCustomizationServiceName + ":" + imagesBindAddress,
+	}
+}
+
+func createImageCustomizationContainer(images *Images, info *ProvisioningInfo) corev1.Container {
+	container := corev1.Container{
+		Name:            "image-customization-controller",
+		Image:           images.ImageCustomizationController,
+		Command:         []string{"/image-customization-controller"},
+		VolumeMounts:    []corev1.VolumeMount{imageRegistriesVolumeMount},
+		ImagePullPolicy: "IfNotPresent",
+		Env: []corev1.EnvVar{
+			{
+				Name:  imagesBindAddress,
+				Value: imagesBindAddressDefault,
+			},
+			setImagePublishUrl(imageCustomizationService, imagesBindAddressDefault, info.Namespace),
+			setIronicBaseUrl(ironicBaseUrl, info),
+			{
+				Name:  ironicAgentImage,
+				Value: images.IronicAgent,
+			},
+			{
+				Name:  imageRegistriesEnvVar,
+				Value: imageRegistriesConfPath,
+			},
+			buildSSHKeyEnvVar(info.SSHKey),
+			pullSecret,
+		},
+		Resources: corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("5m"),
+				corev1.ResourceMemory: resource.MustParse("50Mi"),
+			},
+		},
+	}
+	return container
+}
+
+func newImageCustomizationPodTemplateSpec(info *ProvisioningInfo, labels *map[string]string) *corev1.PodTemplateSpec {
+	containers := []corev1.Container{
+		createImageCustomizationContainer(info.Images, info),
+	}
+
+	tolerations := []corev1.Toleration{
+		{
+			Key:      "node-role.kubernetes.io/master",
+			Effect:   corev1.TaintEffectNoSchedule,
+			Operator: corev1.TolerationOpExists,
+		},
+		{
+			Key:      "CriticalAddonsOnly",
+			Operator: corev1.TolerationOpExists,
+		},
+		{
+			Key:               "node.kubernetes.io/not-ready",
+			Effect:            corev1.TaintEffectNoExecute,
+			Operator:          corev1.TolerationOpExists,
+			TolerationSeconds: pointer.Int64Ptr(120),
+		},
+		{
+			Key:               "node.kubernetes.io/unreachable",
+			Effect:            corev1.TaintEffectNoExecute,
+			Operator:          corev1.TolerationOpExists,
+			TolerationSeconds: pointer.Int64Ptr(120),
+		},
+	}
+
+	return &corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: podTemplateAnnotations,
+			Labels:      *labels,
+		},
+		Spec: corev1.PodSpec{
+			Containers:         containers,
+			HostNetwork:        false,
+			DNSPolicy:          corev1.DNSClusterFirstWithHostNet,
+			PriorityClassName:  "system-node-critical",
+			NodeSelector:       map[string]string{"node-role.kubernetes.io/master": ""},
+			ServiceAccountName: "cluster-baremetal-operator",
+			Tolerations:        tolerations,
+			Volumes: []corev1.Volume{
+				imageRegistriesVolume(),
```

Comment:
Need to also add the `imageVolume()` so we can access the images.

Reply:
Done and mounted it.